food/preprocess.py

```diff
@@ -0,0 +1,183 @@
+from tempfile import TemporaryDirectory
+
+import bs4
+import tqdm
+
+from licensed_pile import logs
+from licensed_pile.write import ShardParallelProcessor
+
+parser = argparse.ArgumentParser(
+    description="Preprocess raw pages in the dolma format."
+)
+parser.add_argument(
+    "--input",
+    default="data/foodista/raw",
+    help="The input version, this directory should be where the `documents` dir lives.",
+)
+parser.add_argument(
+    "--output",
+    default="data/foodista/v0",
+    help="The output version, this directory should be where the `documents` dir will live.",
+)
+# TODO: Respect this flag
+parser.add_argument(
+    "--overwrite",
+    action="store_true",
+    help="Should we overwrite previously processed examples?",
+)
+parser.add_argument(
+    "--debug",
+    action="store_true",
+    help="Should we log when documents are not changed by preprocessing.",
+)
+parser.add_argument(
+    "--processes",
+    type=int,
+    default=mp.cpu_count(),
+    help="Number of processors for multicore.",
+)
+
+# Dolma later sets the log level to error, need to override cls.get_logger() if
+# we want to see info methods.
+logs.configure_logging("dolma.FoodistaParallel")
+
+
+class FoodistaParallel(ShardParallelProcessor):
+    @classmethod
+    def process_example(cls, example, **kwargs):
+        logger = cls.get_logger()
+
+        html = example["text"]
+
+        text, authors, date = parse_page(html, example["id"])
+
+        example["text"] = text
+        example["created"] = date
+        example["metadata"]["authors"] = authors
+
+        return example
+
+
+def parse_page(html, idx):
+    """Convert a page's html to plain text for LLM training."""
+    logger = logs.get_logger("dolma.FoodistaParallel")
+    soup = bs4.BeautifulSoup(html, "html.parser")
+
+    result = []
+    authors = []
+    # Find the title of the article
+    if title := soup.find("h1", id="page-title"):
+        # .get_text() seems to be smart enough to get the text from the inner <a>...</a>
+        title = title.get_text().strip()
+        result.append(title)
+    else:
+        logger.warning(f"Failed to find title for example: {idx}")
+
+    # Find the author's name (which is included in the text) and the user id (which
+    # will be included in the metadata).
+    if author := soup.find("div", class_="pane-node-author"):
+        if user_id := author.find("a", class_="username"):
+            user_id = user_id.get("href")
+        else:
+            logger.warning(
+                f"Failed to find the user_id for the author in example: {idx}"
+            )
+        author = author.get_text().strip()
+        result.append(f"By: {author}")
+        authors.append((author, user_id))
+    else:
+        logger.warning(f"Failed to find author for example: {idx}")
+
+    # Find the date it was published.
+    if date := soup.find("div", class_="pane-node-created"):
+        date = date.get_text().strip()
+        result.append(f"Published: {date}")
```

Comment:
I'm not sure we need the publication date, but if you have a specific reason for including it I'm fine with it.